### ISSUE TITLE
Proposal: Typescript - add ability to run `tsc` only

### ISSUE THREAD

**Issue by hepiyellow** [wix/yoshi#230]
A faster solution then running `yoshi build`. This will improve developer experience.

**Comment by ranyitz** [comment]
Ok, I'll try to explain what yoshi does regarding typescript:

## Testing
You choose a test runner (jest/mocha/jasmine) and each of them takes care of typescript dynamically, for example in mocha you are using `ts-node-register`:
https://github.com/wix/yoshi/blob/132ee11e9140746f8d772a65f6113910939a4aa1/packages/yoshi/src/require-hooks/ts-node-register.js#L1-L4
And I guess that in jest you'll use [`ts-jest`](https://github.com/kulshekhar/ts-jest)

## Bundling
You start from an entry file, webpack build a dependency graph, and transpile all `ts`/`tsx` files using [ts-loader](https://github.com/TypeStrong/ts-loader)

https://github.com/wix/yoshi/blob/132ee11e9140746f8d772a65f6113910939a4aa1/packages/yoshi/src/loaders/typescript.js#L3-L21

## Transpilation
Go over files described in `tsconfig.json` and transpile them, feel free to read about `include`/`files` options - https://www.typescriptlang.org/docs/handbook/tsconfig-json.html

We are [only wrapping tsc](https://github.com/wix/haste/blob/master/packages/haste-task-typescript/src/index.js#L8) and for a regular use case, we are putting the files in `dist` directory (which is the convention for processed files path).

We also supply a method to create 2 bundles, one with `commonjs` target and one with `es2015` (with `import` and `export` for enabling tree shaking).

https://github.com/wix/yoshi/blob/132ee11e9140746f8d772a65f6113910939a4aa1/packages/yoshi/src/commands/build.js#L219-L237

## Storybook
Yoshi does not run storybook, in fact, you need to install it from npm, because it does not ship with yoshi. Yoshi only helps you with the configuration, and supply a webpack configuration which extends storybook config, to enable features like stylable or typescript transpilation.

we use the rules from the common config here:
https://github.com/wix/yoshi/blob/master/packages/yoshi/config/webpack.config.storybook.js#L16 which include the same typescript loader from the bundle section.

## To summarize
If you run tsc, you'll have the "transpile only feature", and you should use `"outDir": "dist"` in your `tsconfig.json` if you want to run it without yoshi.

Hoped I help with explaining how that works.

**Comment by ranyitz** [comment]
As @ronami explained, you can run only `tsc` even without yoshi.

closing
